dbt/adapters/athena/impl.py

```diff
@@ -1314,3 +1309,25 @@ def format_value_for_partition(self, value: Any, column_type: str) -> Tuple[str,
         else:
             # Raise an error for unsupported column types
             raise ValueError(f"Unsupported column type: {column_type}")
+
+    @available
+    def run_optimize_with_partition_limit_catching(self, optimize_query: str) -> None:
+        while True:
+            try:
+                self._run_query(optimize_query, catch_partitions_limit=False)
+                break
+            except OperationalError as e:
+                if "ICEBERG_OPTIMIZE_MORE_RUNS_NEEDED" not in str(e):
+                    raise e
+
+    def _run_query(self, sql: str, catch_partitions_limit: bool) -> Any:
```

Comment:
```suggestion
    def _run_query(self, sql: str, catch_partitions_limit: bool) -> AthenaCursor:
```
seems minor typing enhancement should be added here

Reply:
I tried last time but I got this error:
> dbt/adapters/athena/impl.py:1334: error: Returning Any from function declared to return "AthenaCursor"  [no-any-return]
Found 1 error in 1 file (checked 1 source file)

Reply:
I found a way :)